source/ref_nri/r_image.c

```diff
@@ -372,9 +375,38 @@ static uint8_t *_R_AllocImageBufferCb( void *ptr, size_t size, const char *filen
 	return _R_PrepareImageBuffer( cbinfo->ctx, cbinfo->side, size, filename, linenum );
 }
 
-/*
-* R_ReadImageFromDisk
-*/
+/**
+ *
+ * returns stbi buffer needs to be freed with stbi_image_free
+ **/
+static stbi_uc* __loadFileImageStbi(const char* filepath, int* width, int* height, enum texture_format_e* format) {
+	uint8_t * fileBuffer;
+	size_t size = R_LoadFile( filepath, ( void ** ) &fileBuffer );
+	if( fileBuffer == NULL )
+		return NULL;
+
+	int channelCount;
+	stbi_uc* stbiBuffer = stbi_load_from_memory(fileBuffer, size, 
+																						width,
+																						height,
+																						&channelCount, 0);
+	switch(channelCount) {
+		case 3:
+			*format = R_FORMAT_RGB8_UNORM;
+			break;
+		case 4:
+			*format = R_FORMAT_RGBA8_UNORM;
+			break;
+		default:
+			stbi_image_free(stbiBuffer);
+			stbiBuffer = NULL;
+			ri.Com_Printf(S_COLOR_YELLOW "unhandled channel count: %d", channelCount);
+			break;
+	}
+	R_Free(fileBuffer);
+	return stbiBuffer; 
+}
```

Comment:
might to to generalize this loading logic across the two the details of how to consume the image are up to the implementation.